Tests/IGListAdapterTests.m

```diff
@@ -1067,6 +1067,24 @@ - (void)test_whenSupplementarySourceReturnsNegativeSize_thatAdapterReturnsZero {
     XCTAssertEqual(size.height, 0.0);
 }
 
+- (void)test_whenSectionIndexGreaterThanZero_thatAdapterReturnsSizeForSectionIndexesGreaterThanZero {
+  self.dataSource.objects = @[@1];
+  [self.adapter reloadDataWithCompletion:nil];
+  
+  IGTestSupplementarySource *supplementarySource = [IGTestSupplementarySource new];
+  supplementarySource.collectionContext = self.adapter;
+  supplementarySource.supportedElementKinds = @[UICollectionElementKindSectionFooter];
+  
+  IGListSectionController *controller = [self.adapter sectionControllerForObject:@1];
+  controller.supplementaryViewSource = supplementarySource;
+  supplementarySource.sectionController = controller;
+  
+  const CGSize size = [self.adapter sizeForSupplementaryViewOfKind:UICollectionElementKindSectionHeader
+                                                       atIndexPath:[NSIndexPath indexPathForItem:0 inSection:1]];
```

Comment:
Just adding that the same fix would have to be applied to this method if someone wanted to return a different view according to the section index:
`- (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView viewForSupplementaryElementOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath`

It is also taking the item from the index path. So I can see you're offering people the ability of returning multiple views and multiple sizes for each view, and probably all I need to complete my puzzle is to understand:
1) what should I do in order to be able to receive different indexes in my IGListSupplementaryViewSource implementation?
2) let's say that I succeed to have 2 supplementary views, which are the same class but they have different sizes. So I have indexes 0 and 1 for my supplementary views. How can I code the view index 0 to be used for section index 0, and the view index 1 to be used for the other section indexes?

I'm feeling this will end the discussion. Thanks again!